### ISSUE TITLE
System assigned managed identity permissions are not set during deployment

### ISSUE THREAD

**Issue by EelcoLabordus** [Azure/Enterprise-Scale#903]
**Describe the bug**
During the deployment of ES policy assignments are created but the System assigned managed identity permissions are not set. So some of the policies don't work.

**Steps to reproduce**

1. Go to thew policy assignment
2. Edit assignment
3. Go to remediation (First screen shot)
4. Review and save
5. Edit assignment again and you see the other screenshot

**Screenshots**

![image](https://user-images.githubusercontent.com/75468927/152125400-189a9c37-f21f-4c0b-b245-c19a2a780669.png)

![image](https://user-images.githubusercontent.com/75468927/152125689-50d2a8af-2d16-4e57-92ce-f1dc88f12d95.png)

![image](https://user-images.githubusercontent.com/75468927/152125733-8fadeaaa-d7e5-4d9f-a591-6ee8bd6ef751.png)

![image](https://user-images.githubusercontent.com/75468927/152125849-80002b4f-8fd1-44c2-8e76-87101393a460.png)

**Comment by jtracey93** [comment]
Hi @EelcoLabordus,

Thanks for raising the issue. The policy should indeed work as part of the portal deployment experience and the associated RBAC assignment created at the same scope of the policy assignment. 

However the portal eludes to the fact the Policy Managed Identity isn't assigned RBAC, however it is but we have just granted it Owner and hence the portal thinks that means it isn't assigned, but it is and the policy will indeed work.

You can see this policy assignment you have shown as an example here: https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/managementGroupTemplates/policyAssignments/DINE-SQLEncryptionPolicyAssignment.json and here is the line where we make the RBAC assignment for the policy assignment managed identity https://github.com/Azure/Enterprise-Scale/blob/4ed462c7cf13b4aaa924c60d8e04e51e01c0fea5/eslzArm/managementGroupTemplates/policyAssignments/DINE-SQLEncryptionPolicyAssignment.json#L51

We are working on a workstream to go through all the policies and refine the RBAC roles assigned to them that we hope to complete by the middle of the year.

In summary, if you look under `Owner` RBAC assignments, you should see the policy assignments managed identity assigned. However, the policy portal experience will show a blank field on the assignment identity tab for the "this identity currently has the following permissions" field.

Hope that helps and makes sense 👍

Please let us know if that is the case and we would welcome your feedback as always 👍
